chrome/content/zotero/xpcom/zotero.js

```diff
@@ -2063,3 +2063,79 @@ Zotero.JSON = new function() {
 		return JSON.parse(arg);
 	}
 }
+
+/**
+ * Manages preference panes.
+ */
+Zotero.PreferencePanes = {
+	builtInPanes: Object.freeze([
+		{
+			id: 'zotero-prefpane-general',
+			label: 'zotero.preferences.prefpane.general',
+			image: 'chrome://zotero/skin/prefs-general.png',
+			src: 'chrome://zotero/content/preferences/preferences_general.xhtml',
+			scripts: ['chrome://zotero/content/preferences/preferences_general.js']
+		},
+		{
+			id: 'zotero-prefpane-sync',
+			label: 'zotero.preferences.prefpane.sync',
+			image: 'chrome://zotero/skin/prefs-sync.png',
+			src: 'chrome://zotero/content/preferences/preferences_sync.xhtml',
+			scripts: ['chrome://zotero/content/preferences/preferences_sync.js']
+		},
+		{
+			id: 'zotero-prefpane-export',
+			label: 'zotero.preferences.prefpane.export',
+			image: 'chrome://zotero/skin/prefs-export.png',
+			src: 'chrome://zotero/content/preferences/preferences_export.xhtml',
+			scripts: ['chrome://zotero/content/preferences/preferences_export.js']
+		},
+		{
+			id: 'zotero-prefpane-cite',
+			label: 'zotero.preferences.prefpane.cite',
+			image: 'chrome://zotero/skin/prefs-styles.png',
+			src: 'chrome://zotero/content/preferences/preferences_cite.xhtml',
+			scripts: ['chrome://zotero/content/preferences/preferences_cite.js']
+		},
+		{
+			id: 'zotero-prefpane-advanced',
+			label: 'zotero.preferences.prefpane.advanced',
+			image: 'chrome://zotero/skin/prefs-advanced.png',
+			src: 'chrome://zotero/content/preferences/preferences_advanced.xhtml',
+			scripts: ['chrome://zotero/content/preferences/preferences_advanced.js']
+		},
+		{
+			id: 'zotero-subpane-reset-sync',
+			parent: 'zotero-prefpane-sync',
+			label: 'zotero.preferences.subpane.resetSync',
+			src: 'chrome://zotero/content/preferences/preferences_sync_reset.xhtml',
+			scripts: ['chrome://zotero/content/preferences/preferences_sync.js']
+		}
+	]),
+
+	pluginPanes: [],
+
+	/**
+	 * Register a pane to be displayed in the preferences. The pane XHTML (`src`)
+	 * is loaded as a fragment, not a full document, with XUL as the default
+	 * namespace and (X)HTML tags available under `html:`.
+	 *
+	 * @param {Object} options
+	 * @param {String} options.id Must be unique
+	 * @param {String} [options.parent] ID of parent pane (if provided, pane is hidden from the sidebar)
+	 * @param {String} [options.label] A DTD/.properties key (optional for panes with parents)
+	 * @param {String} [options.rawLabel] A raw string to use as the label if options.label is not provided
+	 * @param {String} [options.image] URI of an icon (displayed in the navigation sidebar)
+	 * @param {String} options.src URI of an XHTML fragment
```

Comment:
Why XUL? Just for the automatic .dtd parsing? Most elements we're using XUL versions of are also available as custom elements in HTML, and, e.g., Mozilla is using HTML for prefs. It doesn't really matter for our panes, but I'm concerned that we don't want to ask plugins to pass XUL here when we're definitely going to want to switch to HTML later. Seems like we could just do the .dtd entity parsing ourselves on the HTML (if there's nothing for that from Mozilla), or even just make plugins use Fluent while they're rewriting things anyway.

Reply:
(And if we want to keep ours as default-XUL for now so that we can merge this, we could have a duplicate adding function that uses default-XHTML for panes coming from `register()`.)